File: hellotoday-app/app/src/main/java/studio/howling/hellotoday/BootReceiver.java
package studio.howling.hellotoday;

import android.content.*;

public class BootReceiver extends BroadcastReceiver {
    @Override public void onReceive(Context context, Intent intent) {
        String action = intent.getAction();
        if (Intent.ACTION_BOOT_COMPLETED.equals(action) || Intent.ACTION_MY_PACKAGE_REPLACED.equals(action)) {
            ReminderScheduler.restoreAll(context);
        }
    }
}

File: opened-again-android/app/src/main/java/com/howling/openedagain/BootReceiver.kt
package com.howling.openedagain

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

/**
 * Re-arms the daily-reminder alarm after a reboot -- AlarmManager alarms
 * are cleared when the device restarts, so without this, the reminder
 * would silently stop firing for anyone who had it turned on until they
 * next opened the app and re-toggled it. See ReminderScheduler.
 */
class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == Intent.ACTION_BOOT_COMPLETED) {
            ReminderScheduler.rescheduleIfEnabled(context)
        }
    }
}
